### PR TITLE
Adds a script the generates and bundle the ios artifacts.

### PR DESCRIPTION
**Pull request by godofredoc**
This is to simplify the migration to engine_v2 recipes and can potentially simplify the existing recipes moving most of the logic to the engine repository.

Bug: https://github.com/flutter/flutter/issues/81855

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: sky/tools/create_full_ios_framework.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+          'Creates Flutter.framework, Flutter.xcframework and '
+          'copies architecture-dependent gen_snapshot binaries to output dir'
+      )
+  )
+
+  parser.add_argument('--dst', type=str, required=True)
+  parser.add_argument('--clang-dir', type=str, default='clang_x64')
+  parser.add_argument('--x64-out-dir', type=str)
+  parser.add_argument('--arm64-out-dir', type=str, required=True)
+  parser.add_argument('--armv7-out-dir', type=str, required=False)
```
Comment (jmagman) [inline sky/tools/create_full_ios_framework.py:40]: Nothing should be using this arg anymore as of https://flutter-review.googlesource.com/c/recipes/+/29060, all `armv7` logic should be removed.  I guess we missed that cleanup in 
https://github.com/flutter/engine/blob/81f540020ad0a0e129f4b8d026ceeff45dcb4fd0/sky/tools/create_macos_gen_snapshots.py#L37
```suggestion
```

File: sky/tools/create_full_ios_framework.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  create_xcframework(location=dst, name='Flutter', frameworks=xcframeworks)
+
+  # Add the x64 simulator into the fat framework
+  subprocess.check_call([
+      'lipo', arm64_dylib, simulator_x64_dylib, '-create', '-output',
+      framework_binary
+  ])
+
+  process_framework(args, framework, framework_binary)
+  generate_gen_snapshot(args, dst, x64_out_dir, arm64_out_dir, armv7_out_dir)
```
Comment (jmagman) [inline sky/tools/create_full_ios_framework.py:156]:
```suggestion
  generate_gen_snapshot(args, dst, x64_out_dir, arm64_out_dir)
```

File: sky/tools/create_full_ios_framework.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  xcframeworks = [simulator_framework, framework]
+  create_xcframework(location=dst, name='Flutter', frameworks=xcframeworks)
+
+  # Add the x64 simulator into the fat framework
+  subprocess.check_call([
+      'lipo', arm64_dylib, simulator_x64_dylib, '-create', '-output',
+      framework_binary
+  ])
+
+  process_framework(args, framework, framework_binary)
```
Comment (jmagman) [inline sky/tools/create_full_ios_framework.py:155]: Can you pull this out into a function called `create_framework` or something similar so it's the same abstraction level as `generate_gen_snapshot` and `zip_archive`?

File: sky/tools/create_full_ios_framework.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+  if args.strip:
+    # copy unstripped
+    unstripped_out = os.path.join(dst, 'Flutter.unstripped')
+    shutil.copyfile(framework_binary, unstripped_out)
+
+    subprocess.check_call(["strip", "-x", "-S", framework_binary])
+
+
+def generate_gen_snapshot(args, dst, x64_out_dir, arm64_out_dir, armv7_out_dir):
```
Comment (jmagman) [inline sky/tools/create_full_ios_framework.py:185]:
```suggestion
def generate_gen_snapshot(args, dst, x64_out_dir, arm64_out_dir):
```

File: sky/tools/create_full_ios_framework.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    _generate_gen_snapshot(
+        os.path.join(arm64_out_dir, args.clang_dir),
+        os.path.join(dst, 'gen_snapshot_arm64')
+    )
+
+  if armv7_out_dir:
+    _generate_gen_snapshot(
+        os.path.join(args.armv7_out_dir, args.clang_dir),
+        os.path.join(dst, 'gen_snapshot_armv7')
+    )
```
Comment (jmagman) [inline sky/tools/create_full_ios_framework.py:199]:
```suggestion
```

File: sky/tools/create_full_ios_framework.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        os.path.join(buildroot_dir, args.x64_out_dir)
+    )
+
+  armv7_out_dir = None
+  if args.armv7_out_dir:
+    armv7_out_dir = (
+        args.armv7_out_dir if os.path.isabs(args.armv7_out_dir) else
+        os.path.join(buildroot_dir, args.armv7_out_dir)
+    )
+
```
Comment (jmagman) [inline sky/tools/create_full_ios_framework.py:80]:
```suggestion
```

File: sky/tools/create_full_ios_framework.py [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+# found in the LICENSE file.
+
+# Generates and zip the ios flutter framework including the architecture
+# dependent snapshot.
+
+import argparse
+import subprocess
+import shutil
+import sys
+import os
```
Comment (zanderso) [inline sky/tools/create_full_ios_framework.py:14]: Please alphabetize imports

**Review by zanderso — APPROVED** [review]
lgtm w/ nit after addressing @jmagman's comments.

File: sky/tools/create_full_ios_framework.py [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+  parser.add_argument('--dst', type=str, required=True)
+  parser.add_argument('--clang-dir', type=str, default='clang_x64')
+  parser.add_argument('--x64-out-dir', type=str)
+  parser.add_argument('--arm64-out-dir', type=str, required=True)
+  parser.add_argument('--armv7-out-dir', type=str, required=False)
+  # TODO(gw280): Remove --simulator-out-dir alias when all recipes are updated
+  parser.add_argument(
+      '--simulator-x64-out-dir', '--simulator-out-dir', type=str, required=True
+  )
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply sky/tools/create_full_ios_framework.py:43]: Done!

File: sky/tools/create_full_ios_framework.py (hunk 7 above)
Comment (godofredoc) [reply sky/tools/create_full_ios_framework.py:14]: done!

File: sky/tools/create_full_ios_framework.py (hunk 1 above)
Comment (jmagman) [reply sky/tools/create_full_ios_framework.py:40]: Argument is still here, are we calling it somewhere?

**Review by jmagman — APPROVED** [review]
LGTM, with nit about removing last `armv7` reference.

**Comment by godofredoc** [comment]
> LGTM, with nit about removing last `armv7` reference.

Removed it, I missed it in the last commit.

**Pull request merged**